### DISCUSSION TITLE
What is the preferred way to include error messages in Markdown?

### DISCUSSION THREAD

**Discussion by GanZhiXiong** [github-linguist/linguist#5927] [Q&A] [answered]
There are similar questions on StackExchange, but no better answers.

- [What is the preferred way to include error messages in Markdown? - Meta Stack Exchange](https://meta.stackexchange.com/questions/165160/what-is-the-preferred-way-to-include-error-messages-in-markdown)
- [Add format for logs/warnings/error messages which allows word-wrapping and preserves newlines - Meta Stack Exchange](https://meta.stackexchange.com/questions/235084/add-format-for-logs-warnings-error-messages-which-allows-word-wrapping-and-prese?rq=1)

**Comment by lildude** [comment]
🤔 this has nothing to do with Linguist and its language detection functionality.

**Reply by GanZhiXiong** [reply]
@lildude 
I am talking about the syntax highlighting of GitHub's Markdown, syntax highlighting is to use Linguist, right?

**Reply by lildude** [reply, marked as answer]
If you're looking to have inline syntax highlighting of GitHub Markdown syntax, you can use the `markdown` or `md` language selector for the code block. 

For example:

```markdown
# Header 1
**bold** _italic_
[google](https://google.com)
```

Markdown has no concept of errors so if you're looking to highlight errors, they'll be specific to the language you're using and then it's up to the upstream grammar to highlight them as you want them.

For example, comments are invalid in JSON, so you could highlight an error by using a comment:

```json
{
  "foo": "bar" // this is an error because it is invalid in JSON and the grammar highlights it.
}
```

**Reply by GanZhiXiong** [reply]
👍🏻 Thank you for your reply. I will try.
